internal/graph/check.go

```diff
@@ -1198,8 +1211,8 @@ func (c *LocalChecker) checkTTU(parentctx context.Context, req *ResolveCheckRequ
 		// If the user is a userset, we will not be able to use the shortcut because the algo
 		// will look up the objects associated with user.
 		if !tuple.IsObjectRelation(tk.GetUser()) {
-			if canShortCircuit, err := typesys.TTUResolvesExclusivelyToDirectlyAssignable(
-				tuple.GetType(object), tuplesetRelation, computedRelation); err == nil && canShortCircuit {
+			if canShortCircuit := typesys.TTUCanFastPath(
```

Comment:
`TTUResolvesExclusivelyToDirectlyAssignable` can be removed in this PR to ensure our replacement path.

I assume you will be building a subsequent PR that would update the userset fast path?

Reply:
> I assume you will be building a subsequent PR that would update the userset fast path?

That would be a logical next step, yes.